docs/reference-manual/native-image/JFR.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: docs
+toc_group: native-image
+link_title: JDK Flight Recorder with Native Image
+permalink: /reference-manual/native-image/JFR/
+---
+
+# JDK Flight Recorder (JFR) with Native Image
+
+JDK Flight Recorder (JFR) is a production time profiling system that is now supported by GraalVM Native Image.
+
+Basically, native images that are built with `-H:+AllowVMInspection` support JFR events written in Java, and users can continue to make use of the  [`jdk.jfr.Event`](https://docs.oracle.com/en/java/javase/11/docs/api/jdk.jfr/jdk/jfr/Event.html) API and JFR itself, with a similar experience to using JFR in the Java HotSpot VM.
+However, to record JFR events at run time, JFR support and JFR recording must be enabled, and this page covers how to start using JFR with native images.
+
+### Current limitations
+
+At the moment, the JFR support is still limited, i.e., some VM-internal events and advanced features such as stack traces or memory leak detection are still missing.
+A subset of JFR features are currently available: custom and system events and disk-based recordings.
+Currently JFR is only supported with native images built on GraalVM JDK 11.
+
+## Compile with JFR
+
+JFR must first be included at image build time. To do so, build an image with the flag `-H:+AllowVMInspection`:
+```shell
+native-image -H:+AllowVMInspection JavaApplication
+```
+
+## Enable JFR
+
+For native images with JFR included, the following flags will be available at run time to enable the system, start a recording, and configure logging:
+
+* `-XX:+FlightRecorder`: use to enable JFR
+* `-XX:StartFlightRecording`: use to start a recording on application's startup
+* `-XX:FlightRecorderLogging`: use to configure the log output for the JFR system
+
+To enable JFR and start a recording, use `-XX:+FlightRecorder` and `-XX:StartFlightRecording` together.
+For example:
+```shell
+./javaapplication -XX:+FlightRecorder -XX:StartFlightRecording="filename=recording.jfr"
+```
+
+## Configure the Recording
+
+Options for -XX:StartFlightRecording
+
+The `StartFlightRecording` flag can optionally be passed a comma separated list of key-value pairs to further configure the recording.
+
+You can pass a comma-separated list of key-value pairs to the `-XX:StartFlightRecording` option to further configure the recording.
```

Comment:
@olyagpl 
I would recommend consolidating the statements on l44, l46 and l48 into one or two. I think there's some repetition otherwise.

Reply:
Done!

Reply:
@olyagpl : I think you didn't push this change yet.